numpy/lib/shape_base.py

```diff
@@ -758,7 +759,54 @@ def array_split(ary, indices_or_sections, axis=0):
     >>> np.array_split(x, 3)
         [array([0.,  1.,  2.]), array([3.,  4.]), array([5.,  6.])]
 
+    >>> x = np.reshape(np.arange(16), (4, 4))
+    >>> np.array_split(x, [3, 3], (0, 1))
+        [array([[0, 1]),
+        array([4, 5]]),
+        array([[2], [6]]),
+        array([[3], [7]]),
+        array([[8, 9]]),
+        array([[10]]),
+        array([[11]]),
+        array([[12, 13]]),
+        array([[14]]),
+        array([[15]])]
+
+    >>> x = np.reshape(np.arange(8), (2, 2, 2))
+    >>> np.array_split(x, [2, 2, 2], (0, 1, 2))
+        [array([[[0]]]),
+        array([[[1]]]),
+        array([[[2]]]),
+        array([[[3]]]),
+        array([[[4]]]),
+        array([[[5]]]),
+        array([[[6]]]),
+        array([[[7]]])]
     """
+    # If axis is a tuple, assume an N-dimensional split.
+    if isinstance(axis, tuple):
+
+        try:
+            len(indices_or_sections)
+        except TypeError:
+            raise ValueError('Axis is a tuple, so indices_or_sections must be of the same length as axis.')
+
+        # Verify there is an index or section for every axis.
+        if len(indices_or_sections) is not len(axis):
```

Comment:
```suggestion
        if len(indices_or_sections) != len(axis):
```
Comparing integers with `is` is not guaranteed to work